python/python/lancedb/embeddings/bedrock.py

```
@@ -15,6 +15,417 @@
+        """
+        Generate an embedding vector for a single image
+
+        Parameters
+        ----------
+        image : Union[str, bytes, PIL.Image.Image]
+            The image to be embedded. If the image is a str, it is treated as a URI.
+            If the image is bytes, it is treated as raw image bytes.
+        """
+        image = self._to_pil(image)
+        image_bytes = self._image_to_bytes(image)
+        base64_image = base64.b64encode(image_bytes).decode('utf-8')
```

Ruff (0.8.2)

Undefined name `base64`

(F821)